### DISCUSSION TITLE
Conditional function won't work after updating to 6.3.0

### DISCUSSION THREAD

**Discussion by sankalp35** [jspsych/jsPsych#1611] [Q&A] [answered]
Hi there,

I updated to the latest version of jsPsych yesterday to fix another question I'd raised, but I realise the conditional function now only works only once at the beginning of a timeline start, even if there are many timeline variables. For the past versions, it used to work at every run of the timeline, not only the first run. I switched the versions back to jsPsych-6, and things worked fine without me tweaking anything. It does seem like a bug, but perhaps it was intentional on your part? I couldn't find any documentation that highlighted this, though!

Hope you can fix it soon?

Best
Sankalp

**Comment by jodeleeuw** [comment]
Hi @sankalp35,

This was an intentional change. The intention is for the `conditional_function` to decide whether the timeline as a whole runs. Can you tell us more about the use case so we can better understand the impact and think about solutions?

**Reply by jodeleeuw** [reply]
Here's the thread where we talked about the changes: https://github.com/jspsych/jsPsych/issues/585

**Reply by sankalp35** [reply]
Hi @jodeleeuw 
I see that this change is quite recent. My use case is I'm running a timeline which I want to loop until a fixed time. The conditional function, at the start of every repetition of the timeline, checks whether the elapsed time is less than the allocated time. If true, then the timeline should run, if not then it should stop. Basically, I am checking for the elapsed time on every run. 

How do you suggest I go about solving this? Is this a potential impact you had considered?

**Comment by jodeleeuw** [comment, marked as answer]
How about adding a `call-function` plugin to the end of the timeline:

```js
var end_timeline_check = {
	type: 'call-function',
	func: function(){
		if(time_elapsed > time_allowed){
			jsPsych.endCurrentTimeline();
		}
	}
}
```

**Reply by jodeleeuw** [reply]
You could also put this inside the `on_finish` function of the last trial in your timeline if you don't want to add the `call-function` trial.

**Reply by sankalp35** [reply]
Thanks for these solutions, can't see why they won't work!
